### DISCUSSION TITLE
RabbitMQ Auto Cluster Setup (using DNS Discovery) in Docker Swarm

### DISCUSSION THREAD

**Discussion by margamraviteja** [rabbitmq/rabbitmq-server#9192] [Other] [answered]
### Describe the bug

Hi, I’m using rabbitmq 3.12-management-alpine docker image and I’m trying to setup rabbitmq cluster of 5 nodes in docker swarm with rabbit peer discovery dns and I was getting Could not auto-cluster with node {badrpc,nodedown}
![rabbitmq_issue](https://github.com/rabbitmq/rabbitmq-server/assets/10586267/11fe0bf3-ad48-4530-af00-7d745e09022c)

### Reproduction steps

Using docker compose yaml file to setup a RabbitMQ Auto Cluster setup in docker swarm

### Expected behavior

I was expecting the cluster starts with the configured number of container replicas in clusters instead of individual standalone rabbitmq servers

### Additional context

I was using docker compose deploy endpoint mode dnsr. I was using same erlang cookie in all the rabbitmq containers

**Comment by michaelklishin** [comment, marked as answer]
`nodedown` means that this node tried to contact another and could not (it then tried another few discovered nodes to no avail). The discovery part did work as expected, see the log line that says "All discovered existing cluster peers".

Network connectivity between nodes in a Swarm is something that RabbitMQ does not control. Something is missing around your networking setup.

**Reply by margamraviteja** [reply]
Thanks. This confirms the rabbitmq cluster setup was good. I will check the network connectivity between the nodes
